src/xr/xr-anchor.js

```diff
@@ -0,0 +1,104 @@
+import { EventHandler } from '../core/event-handler.js';
+
+import { Vec3 } from '../math/vec3.js';
+import { Quat } from '../math/quat.js';
+
+/**
+ * @class
+ * @name XrAnchor
+ * @classdesc Anchor provides position and rotation, that is updated by underlying AR system, that tries to better persist an anchor relative to evolving understanding of a real world.
+ * @description Anchor provides position and rotation, that is updated by underlying AR system, that tries to better persist an anchor relative to evolving understanding of a real world.
+ * @param {XrAnchors} anchors - Anchors manager.
+ * @property {object} xrAndhor native XRAnchor object that is provided by WebXR API.
+ */
+class XrAnchor extends EventHandler {
+    constructor(anchors, xrAnchor) {
+        super();
+
+        this._anchors = anchors;
+        this._xrAnchor = xrAnchor;
+
+        this._position = new Vec3();
+        this._rotation = new Quat();
+    }
+
+    /**
+     * @event
+     * @name XrAnchor#remove
+     * @description Fired when {@link XrAnchor} is removed.
+     * @example
+     * // once anchor is removed
+     * anchor.once('remove', function () {
+     *     // destroy its related entity
+     *     entity.destroy();
+     * });
+     */
+
+    /**
+     * @event
+     * @name XrAnchor#change
+     * @description Fired when {@link XrAnchor}'s position and/or rotation is changed.
+     * @example
+     * anchor.on('change', function () {
+     *     // anchor has been updated
+     *     entity.setPosition(anchor.getPosition());
+     *     entity.setRotation(anchor.getRotation());
+     * });
+     */
+
+    /**
+     * @function
+     * @name XrAnchor#remove
```

Comment:
I think it's normally best to pair `create` with `destroy` and `add` with `remove`. But here, to add an anchor, you call `create` on the manager, but to remove, you call `remove` on the anchor instance.

Reply:
Should `remove` be moved onto the manager?

Reply:
I will use create+destroy (on instance).
As `add` I believe requires something provided to it, like adding an asset to assets registry. Here we.calling a create of an anchor. And then we can destroy it.